### PR TITLE
Create initial documentation structure

### PR DESCRIPTION
**Pull request by Timoses**

### CODE REVIEW

**Comment by LexiconCode** [comment]
Using the readme instructions I had the following issues installing.
```
C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\Caster>cd docs

C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\Caster\docs>pip install -r requirements.txt
Processing c:\users\main\desktop\caster-test-kaldi-py3-64bit\caster
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\python.exe' 'C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\site-packages\pip\_vendor\pep517\_in_process.py' get_requires_for_build_wheel 'C:\Users\Main\AppData\Local\Temp\tmpfp8r9j8h'
       cwd: C:\Users\Main\AppData\Local\Temp\pip-req-build-id_sbmzq
  Complete output (38 lines):
  Traceback (most recent call last):
    File "C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\site-packages\pip\_vendor\pep517\_in_process.py", line 280, in <module>
      main()
    File "C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\site-packages\pip\_vendor\pep517\_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\site-packages\pip\_vendor\pep517\_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\build_meta.py", line 149, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\build_meta.py", line 130, in _get_build_requires
      self.run_setup()
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\build_meta.py", line 253, in run_setup
      super(_BuildMetaLegacyBackend,
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\build_meta.py", line 145, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 6, in <module>
      setuptools.setup(
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\distutils\core.py", line 108, in setup
      _setup_distribution = dist = klass(attrs)
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\dist.py", line 423, in __init__
      _Distribution.__init__(self, {
    File "C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\distutils\dist.py", line 292, in __init__
      self.finalize_options()
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools\dist.py", line 695, in finalize_options
      ep(self)
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools_scm\integration.py", line 61, in infer_version
      dist.metadata.version = _get_version(config)
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools_scm\__init__.py", line 173, in _get_version
      parsed_version = _do_parse(config)
    File "C:\Users\Main\AppData\Local\Temp\pip-build-env-ye5qnlk5\overlay\Lib\site-packages\setuptools_scm\__init__.py", line 134, in _do_parse
      raise LookupError(
  LookupError: setuptools-scm was unable to detect version for 'C:\\Users\\Main\\AppData\\Local\\Temp\\pip-req-build-id_sbmzq'.

  Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

  For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
  ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\python.exe' 'C:\Users\Main\Desktop\Caster-Test-Kaldi-py3-64bit\WPy64-3860\python-3.8.6.amd64\lib\site-packages\pip\_vendor\pep517\_in_process.py' get_requires_for_build_wheel 'C:\Users\Main\AppData\Local\Temp\tmpfp8r9j8h' Check the logs for full command output.
```

Manually fixed with the following
`pip install Sphinx sphinx-rtd-theme sphinx-autobuild`

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sphinx
```
Comment (LexiconCode) [inline docs/requirements.txt:1]: Should be for PIP packages to be installed via `pip install -r requirements.txt`
```
Sphinx 
sphinx-rtd-theme 
sphinx-autobuild
```

**Comment by LexiconCode** [comment]
When running `make auto`.

```
Sphinx error:
Builder name auto not registered or available through entry point
```

**Comment by LexiconCode** [comment]
In addition I've tried `sphinx-autobuild` manually but without further knowledge of sphinx project structure it's difficult to troubleshoot.

File: docs/requirements.txt (hunk 1 above)
Comment (Timoses) [reply docs/requirements.txt:1]: Fixed in https://github.com/Timoses/Caster/pull/17/commits/919b52d4d06c98d24e46de2de3b04a8decc89bf4

**Comment by Timoses** [comment]
Yes, castervoice package must be installed. Apparently Windows python does not like some 

Try the new documentation building in README.md:
```
pip install . -r docs/requirements.txt
cd docs
make html
```


----

> Include .\ to make a cross compatible between cmd and powershell for executing the bat.
.\make auto

I'm not super familiar with Windows Powershell etc. But I believe it is the same that when executing a program plainly as in `make` uses the program in the PATH, whereas `./make` (presumably also `.\make` would use the make program in the current directory.
However the user wishes to execute `make` should be in his or her responsibility.

**Comment by LexiconCode** [comment]
Now that we have the package caster installing

> When running `.\make auto`.
> 
> ```
> Sphinx error:
> Builder name auto not registered or available through entry point
> ```

`sphinx-autobuild source /_build/html` must be the equivalent of `make auto`

**Comment by Timoses** [comment]
Not sure what kind of `make` is on Windows by default. There seem to be various options:
http://gnuwin32.sourceforge.net/packages/make.htm
https://stackoverflow.com/questions/32127524/how-to-install-and-use-make-in-windows

Since the docs are meant to be hosted online I don't think it's important to focus on a cross-compatible docs Makefile at this point.
You can always use Linux in a VM ; ).
Or run it in a container, e.g. with Docker, and forward the port:
```
docker run -it --name caster -p 8000:8000 -v $(pwd):/caster python bash -c "cd /caster && pip install . -r docs/requirements.txt && cd docs && sphinx-autobuild --host 0.0.0.0 source _build/html"
```

**Comment by LexiconCode** [comment]
All we don't have to go that far with the VM is just a little bit different with Windows for instruction for building the documentation. `sphinx-autobuild` does the heavy lifting. Like you said it's not a high priority at this point. The initial doc structure looks good to me I something will continue to develop as the project advances.

**Pull request merged**